src/mito2/src/compaction/twcs.rs

```diff
@@ -88,7 +88,7 @@ impl TwcsPicker {
             // because after compaction there will be no overlapping files.
             let filter_deleted = !files.overlapping && found_runs <= 2 && !self.append_mode;
             if found_runs == 0 {
-                return output;
+                continue;
```

Comment:
We may need to limit the number of output. Otherwise, the region may have a large compaction task after fixing this.